prog/weaveutil/datapath.go

```diff
@@ -31,9 +54,81 @@ func deleteDatapath(args []string) error {
 	return odp.DeleteDatapath(args[0])
 }
 
+// Checks whether a datapath can be created by actually creating and destroying it
+func checkDatapath(args []string) error {
+	if len(args) != 1 {
+		cmdUsage("check-datapath", "<datapath>")
+	}
+
+	dpname := args[0]
+
+	odpSupported, err := odp.CreateDatapath(dpname)
+	if err != nil {
+		return err
+	}
+	if !odpSupported {
+		return fmt.Errorf("kernel does not have ODP support")
+	}
+
+	return odp.DeleteDatapath(args[0])
+}
+
 func addDatapathInterface(args []string) error {
 	if len(args) != 2 {
 		cmdUsage("add-datapath-interface", "<datapath> <interface>")
 	}
 	return odp.AddDatapathInterface(args[0], args[1])
 }
+
+// Helpers
+
+func checkMTU(dpname string, mtu int) {
+	var (
+		vpid   libodp.VportID
+		vpname string
+	)
+
+	// Create a dummy vxlan vport
+	for i := 0; i < 5; i++ {
+		portno, err := getUDPPortNo()
+		if err != nil {
+			return
+		}
+		if vpid, vpname, err = odp.CreateVxlanVport(dpname, "vxlantest", portno); err == nil {
+			defer func() {
+				if err := odp.DeleteVport(dpname, vpid); err != nil {
+					fmt.Fprintf(os.Stderr, "unable to remove unused %q vxlan vport: %s\n", vpname, err)
+				}
+			}()
+			break
+		} else {
+			if errno, ok := err.(syscall.Errno); ok && errno == syscall.EADDRINUSE {
+				continue
+			}
+		}
+	}
+	// Couldn't create the vport, skip the check
+	if vpname == "" {
+		return
+	}
+
+	// Setting >1500 MTU on affected host' vport should fail with EINVAL
+	if err := wnet.SetMTU(vpname, mtu); err != nil {
+		if errno, ok := err.(syscall.Errno); ok && errno == syscall.EINVAL {
+			fmt.Fprintf(os.Stderr, "WARNING: fastdp won't work with %d MTU potentially due to 4.3 or 4.4 kernel version\n", mtu)
```

Comment:
No, it's not. See https://github.com/weaveworks/weave/pull/2373#discussion_r66999339

Also, I thought we said we'd make this an *error*?